Automattic/wp-calypso
# Store: Use Checklist component for dashboard.

This branch closes #21322 and also fixes #18825 by removing the existing dashboard checklist component, and replaces it with the `<Checklist />` component instead: 

__Before__
![dash-before](https://user-images.githubusercontent.com/22080/37064767-1ba9253a-2153-11e8-9c6a-e183d1fd704c.png)

__After__
![dashboard-setup](https://user-images.githubusercontent.com/22080/37064824-557bdb5e-2153-11e8-8549-378dbe2860a3.png)

Some of the noticeable changes of using the `<Checklist />` component is as you finish items in the checklist, they move to the top of the list, instead of staying in their original spot. Also the 'DO IT' button is kind of _interesting_. We could consider updating the base component to allow for a custom button text:

![in-progress](https://user-images.githubusercontent.com/22080/37064865-8190dd5c-2153-11e8-99ec-67ed151f70c6.png)

__To Test__
The easiest way to test out the setup checklist is to use the REST API console [https://developer.wordpress.com/docs/api/console/](https://developer.wordpress.com/docs/api/console/) to reset the preferences to the initial install state using the `/sites/$site/calypso-preferences/woocommerce` endpoint. A fresh install has values that look like so:

![new-site-settings](https://user-images.githubusercontent.com/22080/37064936-be423fd4-2153-11e8-93ea-ecb405eb2784.png)

Once you have reset your settings, you can then interact with the setup checklist and verify it works as expected.
